static_frame/core/frame.py

```diff
@@ -5097,13 +5103,19 @@ def blocks() -> tp.Iterator[np.ndarray]:
         else:
             block_gen = blocks
 
-        if self._columns.depth > 1:
-            raise ErrorInitFrame('cannot unset index with a columns with depth greater than 1')
+        # self._columns._blocks may be None until array cache is updated.
+        self._columns._update_array_cache()
+        column_blocks = self._columns._blocks._blocks
```

Comment:
Here I would follow pattern as above and check the recache status first:

```suggestion
        if self._columns._recache:
            self._columns._update_array_cache()
        column_blocks = self._columns._blocks._blocks
```